app/models/retrieval.rb

```diff
@@ -64,24 +64,34 @@ def self.process(id, token, env, base_url, access_token)
       retrieval.collections.each do |collection|
         params = Rack::Utils.parse_nested_query(collection['params'])
         params.reject! { |p| ['datasource', 'short_name'].include? p }
-        params.merge!(page_size: 2000, page_num: 1)
+
+        results_count = test_query_params(client, params, token)
+        page_num = 0
+        page_size = 2000
 
         access_methods = collection['serviceOptions']['accessMethod']
         access_methods.each do |method|
           begin
             if method['type'] == 'order'
-              order_response = client.create_order(params,
-                                                   method['id'],
-                                                   method['method'],
-                                                   method['model'],
-                                                   user_id,
-                                                   token,
-                                                   client,
-                                                   access_token)
-              method[:order_id] = order_response[:order_id]
-              method[:dropped_granules] = order_response[:dropped_granules]
-              Rails.logger.info "Granules dropped from the order: #{order_response[:dropped_granules].map { |dg| dg[:id] }}"
+              # TODO put upper limit of 1000000 granules
+              until page_num * page_size > results_count do
+                page_num += 1
+                params.merge!(page_size: page_size, page_num: page_num)
+                order_response = client.create_order(params,
+                                                     method['id'],
+                                                     method['method'],
+                                                     method['model'],
+                                                     user_id,
+                                                     token,
+                                                     client,
+                                                     access_token)
+                method[:order_id] ||= []
+                method[:order_id] << order_response[:order_id]
+                method[:dropped_granules] = order_response[:dropped_granules]
```

Comment:
should this be ` method[:dropped_granules] << order_response[:dropped_granules]`? If it is an ASTER order and is split into 2 orders. The first order drops, say 130 granules and the second one drops 500 granules. A total of 630 granules should be shown in the dropped granules panel.

I think you'd better let ASTER (LP DAAC?) know before you submit orders with large numbers of granules. I was black-listed last time when I did this...